santosfamilyfoundation/RuinaMain
# Feat/casey j may/validation customization

## Validation Customization
This feature should allow the spreadsheet to update the type, values, and error messages associated with the validation in individual questions. 

### Testing

1. First, edit this copy of the spreadsheet: https://docs.google.com/spreadsheets/d/14XByy6qqHQagWF2f2QkS5S9UeYNF-PqRIkBI1z3tqYo/edit?usp=sharing. Go to the driver's sheet, and either add some validation comments or edit the existing ones. Make sure to use a few selection questions and a few text fields.  HINT: The documentation of the format is not great yet, but just be sure to match the formatting of the existing validation rows.
2. After you've finished editing it, load this spreadsheet into the app. 
3. Create a new report, add a vehicle.
4. Go into the driver form. Test out each of the questions you edited the validation on, and make sure the validation is functioning as intended. 
5. Do some general experimentation. Make sure the runtime is unaffected, try out some fields that you didn't edit as controls, and try and break the app. Put about 5 minutes into this.
6. Confirm there are no leftover print statements. (There shouldn't be)

Thank you!
